### PR TITLE
Apply clang-format (Google-style)

### PR DESCRIPTION
**Pull request by jahfer**
Assuming `clang-format` is installed, `bin/fmt` will format the C files in the repo to match Google's recommended formatting.

Aside from the formatting, this introduces the `bin/fmt` script and the `dev fmt` command (which executes the aformentioned script). No other changes.

### CODE REVIEW

File: ext/rotoscope/rotoscope.h [hunk 1]
```diff
@@ -52,10 +52,13 @@ typedef struct
...
-typedef struct
-{
+typedef struct {
   VALUE name;
   const char *method_level;
 } rs_class_desc_t;
 
+// extconf.rb has already asserted we are on POSIX, this is just to make the
+// compiler happy
+int fileno(FILE *stream);
```
Comment (dylanahsmith) [inline ext/rotoscope/rotoscope.h]: Why is this needed now when it wasn't before?

File: ext/rotoscope/extconf.rb [hunk 2]
```diff
@@ -2,4 +2,6 @@
 require "mkmf"
 
 $CFLAGS << ' -std=c99 -Wall -Werror -Wno-declaration-after-statement'
+
+have_macro('_POSIX_SOURCE')
```
Comment (dylanahsmith) [inline ext/rotoscope/extconf.rb]: It doesn't look like this does anything.  [have_macro](http://ruby-doc.org/stdlib-2.3.3/libdoc/mkmf/rdoc/MakeMakefile.html#method-i-have_macro) returns a value, but you aren't checking it and it doesn't document doing any side-effects

**Comment by jahfer** [comment]
@dylanahsmith You're faster than me, this wasn't ready for others' eyes yet ;) The problem with `fileno` actually stemmed from `ruby.h` (and the other ruby files) being included _after_ `stdio.h`, and for whatever reason that started triggering [this compiler error](https://circleci.com/gh/Shopify/rotoscope/189).

Having moved the Ruby files ahead of `stdio.h` seems to have sorted out the problem. I verified it by running a Docker container that hit the same issue as CircleCI, and it runs smoothly now.

**Comment by jahfer** [comment]
Github messed up the PR-closing logic here…new PR opened at #33.
